### PR TITLE
Work

### PR DESCRIPTION
**Pull request by 770menahem**

### CODE REVIEW

File: src/api/entity.ts [hunk 1]
```diff
@@ -2,18 +2,26 @@ import axios from 'axios';
 import { krtflEntity, entity } from '../types/entityType';
 import config from '../config/env.config';
 import { getResData } from './getResData';
+import getToken from '../auth/spike';
 
 export const entityApi = {
   create: async (entity: entity) => {
-    return await getResData(axios.post(`${config.krtflApi}/entities`, entity));
+    return await getResData(axios.post(`${config.krtflApi}/entities`, entity, await getToken()));
```
Comment (NoamYes) [inline src/api/entity.ts:9]: couldn't getToken could be included in an axios middleware?

File: src/api/getResData.ts [hunk 2]
```diff
@@ -1,11 +1,17 @@
 import { AxiosResponse } from 'axios';
-import { logError } from './../logger/logger';
+import { logWarn } from './../logger/logger';
 
 export const getResData = async (req: Promise<AxiosResponse<any>>) => {
   try {
-    return (await req).data;
+    const res = await req;
```
Comment (NoamYes) [inline src/api/getResData.ts:6]: req is misleading a littlebit

File: src/app.ts [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+import { DailyRun } from './redis/DailyRun';
+import redisClient from './redis/redis';
 
-(async () => {
-  await connectRabbit();
-})();
+const start = () => {
+  redisClient(async () => {
+    await connectRabbit();
+    DailyRun.getInstance().start();
```
Comment (NoamYes) [inline src/app.ts:8]: a daily run probably means a recovery once in a while for missing connections?

File: src/redis/DailyRun.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  public async start() {
+    logInfo(`Daily run scheduled to ${this.hour}:${this.minute}`);
+    this.runFunc = schedule.scheduleJob({ hour: this.hour, minute: this.minute }, () => {
+      logInfo('Daily run is starting');
+      runAll();
+    });
+  }
+
+  public changeRunTime(hour: number, minute: number) {
```
Comment (NoamYes) [inline src/redis/DailyRun.ts:35]: one would use this or the running configuration time should be included in the config?

File: src/redis/redis.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    console.log('Redis connected (di to entity)');
+    cb();
+  });
+
+  client.on('error', (err) => {
+    console.error('Redis Error: ' + err);
+  });
+
+  delValue = async (key: string) => client.del(`DI_TO_ENTITY ${key}`);
+  pushToArray = async (key: string, value: string) => client.lpush(`DI_TO_ENTITY ${key}`, value);
```
Comment (NoamYes) [inline src/redis/redis.ts:23]: DI_TO_ENTITY prefix added for useful search?

File: src/redis/redis.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  getArray = async (key: string) => {
+    return promisify(client.lrange).bind(client)(`DI_TO_ENTITY ${key}`, 0, -1);
+  };
+  getAllKeys = async () => {
```
Comment (NoamYes) [inline src/redis/redis.ts:27]: this function should remove the prefix?

File: src/service/entity/saveEntity.ts [hunk 7]
```diff
@@ -12,17 +13,18 @@ export const insertEntity = async (entity: entity) => {
 
   if (!krtflEntity) {
     krtflEntity = await entityApi.create(entity);
+    logInfo('Entity created successfully', krtflEntity?.id);
+    entityEvent(entity.identityCard || entity.personalNumber || entity.goalUserId!);
```
Comment (NoamYes) [inline src/service/entity/saveEntity.ts:17]: what does it exactly do? trying to connect every di that should be attached? consider more approachable name

File: src/service/rgb/saveDI.ts [hunk 8]
```diff
@@ -35,7 +35,7 @@ export const insertDI = async (di: di) => {
   if (entityIdentifier) {
     await connectDiToEntity(krtflDI, entityIdentifier);
```
Comment (NoamYes) [inline src/service/rgb/saveDI.ts:36]: should also be redis implemented?

**Review by NoamYes — COMMENTED** [review]
Nice work
will continue another time

File: src/app.ts (hunk 3 above)
Comment (RoeiOren) [reply src/app.ts:8]: Yes. Once a day

File: src/redis/redis.ts (hunk 5 above)
Comment (RoeiOren) [reply src/redis/redis.ts:23]: For creating and simulating a "collection" in Redis and separate the (key, value) pairs of DIs waiting for a connection from all the other (key, value) pairs in the same Redis instance(like ('accessToken', '3216531876258732') and etc.)

File: src/redis/redis.ts (hunk 6 above)
Comment (RoeiOren) [reply src/redis/redis.ts:27]: this function returns the array value of a key(entitiy's identifier). It automatically adds the prefix for the given identifier

File: src/service/entity/saveEntity.ts (hunk 7 above)
Comment (RoeiOren) [reply src/service/entity/saveEntity.ts:17]: When the entity is added, if his identifier exists in Redis DB, the value of his identifier key(arrays of DIs waiting to connect him) being pulled, iterated over the DIs, and connecting them to the entity.

File: src/redis/connectDiToEntityRedis.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  } else {
+    pushToArray(entityId, diId);
+  }
+};
+
+export const entityEvent = async (entityId: string) => {
+  const data = await getArray(entityId);
+
+  if (data.length > 0) {
+    data.forEach((diId) => connectDiToEntityApi(entityId, diId));
```
Comment (NoamYes) [inline src/redis/connectDiToEntityRedis.ts:19]: necessary awaiting before deleting entityId from redis?

**Review by NoamYes — COMMENTED** [review]
boom

**Pull request merged**